config/http_config.go

```diff
@@ -714,6 +750,8 @@ type TLSConfig struct {
 	ServerName string `yaml:"server_name,omitempty" json:"server_name,omitempty"`
 	// Disable target certificate validation.
 	InsecureSkipVerify bool `yaml:"insecure_skip_verify" json:"insecure_skip_verify"`
+	// Minimum TLS version.
+	MinVersion TLSVersion `yaml:"min_version,omitempty"`
```

Comment:
it is missing here the json tag

```suggestion
MinVersion TLSVersion `yaml:"min_version,omitempty" json:"min_version,omitempty"`
```

Reply:
if `MinVersion` is not mandatory, don't you want to provide a default value ?

You could do it by overriding the Unmarshal's method.

Reply:
No, I want to use go default. I think we should let them define the default while providing or users a way to divert if really needed.

Reply:
mmm ok. And I guess in Prometheus you will define that the default / Minimum version is the previous one used, right ?

Reply:
Nope, the plan is to mark it as a CHANGE in the changelog.

I think out users will benefit from defaults that go runtime provides. We should probably not make the call.

Possibly it can break some users, but they have the option to opt-out this change by explicitly setting tls1.0 as supported version.

Reply:
I can agree on this assumption and I agree it's better to use the default tls version supported by Go.
I'm just wondering how much it will impact the users.
Let's see if someone answers to your email.